circuits/cpp/src/aztec3/circuits/hash.hpp

```diff
@@ -137,6 +138,24 @@ typename NCT::fr silo_nullifier(typename NCT::address contract_address, typename
     return NCT::compress(inputs, aztec3::GeneratorIndex::OUTER_NULLIFIER);
 }
 
+template <typename NCT> typename NCT::fr compute_block_hash(typename abis::GlobalVariables<NCT> globals,
+                                                            typename NCT::fr private_data_tree_root,
+                                                            typename NCT::fr nullifier_tree_root,
+                                                            typename NCT::fr contract_tree_root,
+                                                            typename NCT::fr l1_to_l2_data_tree_root,
+                                                            typename NCT::fr public_data_tree_root)
+{
+    using fr = typename NCT::fr;
+
+    std::vector<fr> const inputs = {
+        globals.hash(),     private_data_tree_root,  nullifier_tree_root,
+        contract_tree_root, l1_to_l2_data_tree_root, public_data_tree_root,
+    };
+
+    // TODO(Maddiaa): does this need an index?
```

Comment:
We need to have a bigger conversation about domain separators (hash indices). With our current pedersen implementation we have a limit to the number of hash indices we can have. It causes friction every time we breach that limit. We also need to consult with the crypto team to understand where exactly domain separators are essential. And we need to implement the alternative pedersen hash which should reduce friction around generators.

For now, I'm happy for there to be a TODO signposting this, because the sandbox doesn't claim to be secure yet.